File: database_connection/task_list/task_list_java/src/main/java/com/repository/TaskRepository.java
package com.repository;

import java.util.List;
import java.util.Optional;

import com.model.Task;

public interface TaskRepository {
  void insert(Task task);

  void update(int id, Task task);

  void deleteById(int id);

  Optional<Task> findById(int id);

  List<Task> findAll();
}

File: database_connection/task_list/task_list_python_venv/repository/task_repository.py
from model.task import Task


class TaskRepository:
  def insert(self, task: Task):
    pass

  def update(self, id: int, task: Task):
    pass

  def delete_by_id(self, id: int):
    pass

  def find_by_id(self, id: int) -> Task:
    pass

  def find_all(self) -> list[Task]:
    pass
